File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/server/ChannelAvailableAutoJoinPacket.java
package pe.nanamochi.banchus.packets.server;

import java.io.IOException;
import java.io.OutputStream;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.core.ServerPacket;
import pe.nanamochi.banchus.io.IDataWriter;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ChannelAvailableAutoJoinPacket implements ServerPacket {
  private String realname;
  private String topic;
  private int userCount;

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_CHANNEL_AVAILABLE_AUTOJOIN;
  }

  @Override
  public void write(IDataWriter writer, OutputStream stream) throws IOException {
    writer.writeString(stream, realname);
    writer.writeString(stream, topic);
    writer.writeInt32(stream, userCount);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/ChannelAvailableAutoJoinPacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.core.ServerPacket
import pe.nanamochi.banchus.io.DataWriter

class ChannelAvailableAutoJoinPacket(
    var realName: String = "",
    var topic: String = "",
    var userCount: Int = 0,
) : ServerPacket() {
    override val type = PacketType.BANCHO_CHANNEL_AVAILABLE_AUTOJOIN

    override fun write(writer: DataWriter, out: OutputStream) {
        writer.writeString(out, realName)
        writer.writeString(out, topic)
        writer.writeInt32(out, userCount)
    }
}
